test/lib/next-test-utils.ts

```diff
@@ -523,10 +523,28 @@ export async function killProcess(
   })
 }
 
+export function isAppRunning(instance: ChildProcess) {
+  if (!instance?.pid) {
+    return false
+  }
+
+  try {
+    process.kill(instance.pid, 0)
+    return true
+  } catch {
+    return false
+  }
+}
+
 // Kill a launched app
 export async function killApp(instance: ChildProcess) {
   if (instance && instance.pid) {
     await killProcess(instance.pid)
+    try {
+      await waitForCondition(2000, () => !isAppRunning(instance))
+    } catch {
+      await killProcess(instance.pid, 'SIGKILL')
+    }
```

Comment:
Because it's throwing `SIGTERM` by default, the apps will run indefinitely. Maybe we just have it throw `SIGKILL` by default instead and then adding a separate test for the clean shutdown would look like sending a `SIGTERM` instead and making sure it shuts down properly.

Since `SIGTERM` now waits for the app to shut down properly, it'll still require at least that change to the util